### PR TITLE
tools: add_room changed to cut_wall

### PR DESCRIPTION
**Pull request by RobotCaleb**
cuts out 2x2x1 oriented to your view on the wall

### CODE REVIEW

File: src/ship_space.cc [hunk 1]
```diff
@@ -825,6 +826,7 @@ block *ship_space::get_block_neighbor(glm::ivec3 block, enum surface_index si) {
 }
 
 void ship_space::cut_out_cuboid(glm::ivec3 mins, glm::ivec3 maxs, surface_type type) {
+    printf("  %s\n  %s\n---------\n", glm::to_string(mins).c_str(), glm::to_string(maxs).c_str());
```
Comment (RobotCaleb) [inline src/ship_space.cc]: Oops, I'll get rid of this

Comment (RobotCaleb) [reply src/ship_space.cc]: Fixed

File: src/common.h [hunk 2]
```diff
@@ -130,14 +130,13 @@ surface_index_to_normal(int index)
...
 
 static inline glm::mat4
-mat_rotate_mesh(glm::vec3 pt, glm::vec3 dir) {
+mat_rotate_mesh(glm::vec3 pt, glm::vec3 dir, glm::vec3 up = glm::vec3{0, 0, 1}) {
     auto abs_normal = glm::abs(dir);
-    glm::vec3 temp_up(0, 0, 1);
     if (abs_normal.z > abs_normal.y && abs_normal.z > abs_normal.x) {
         /* avoid degeneracy at the `poles` */
-        temp_up = glm::vec3(0, 1, 0);
+        up = glm::vec3(0, 1, 0);
```
Comment (chrisforbes) [inline src/common.h]: Hang on, this isn't right.

Comment (chrisforbes) [reply src/common.h]: The condition here only makes sense if `up` has the default value.

File: src/tools/cut_wall.cc [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            door_center + u + v,
+        };
+
+        // cut out door
+        ship->cut_out_cuboid(glm::min(doors[0], doors[3]), glm::max(doors[0], doors[3]), surface_wall);
+
+        ship->validate();
+    }
+
+    glm::vec3 get_closest_alignment(const std::vector<glm::vec3> &cand, glm::vec3 v, glm::vec3 normal) {
```
Comment (chrisforbes) [inline src/tools/cut_wall.cc]: We didnt end up using this

File: src/common.h [hunk 4]
```diff
@@ -132,7 +132,7 @@ surface_index_to_normal(int index)
 static inline glm::mat4
 mat_rotate_mesh(glm::vec3 pt, glm::vec3 dir, glm::vec3 up = glm::vec3{0, 0, 1}) {
     auto abs_normal = glm::abs(dir);
-    if (abs_normal.z > abs_normal.y && abs_normal.z > abs_normal.x) {
+    if (up == glm::vec3{0, 0, 1} && abs_normal.z > abs_normal.y && abs_normal.z > abs_normal.x) {
```
Comment (chrisforbes) [inline src/common.h]: This doesn't work. We also don't need this change at all.

Comment (chrisforbes) [reply src/common.h]: I'll clean this up and squash before merging.

**Pull request merged**